### ISSUE TITLE
gallery directive fails after ``nikola clean``

### ISSUE THREAD

**Issue by devurandom** [getnikola/nikola#1251]
```
# nikola clean
# nikola build
[…]
########################################
TaskError - taskid:render_posts:cache/posts/MY_POST.html
PythonAction Error
Traceback (most recent call last):
  File "/usr/lib/python3.3/site-packages/doit/action.py", line 343, in execute
    returned_value = self.py_callable(*self.args, **kwargs)
  File "/usr/lib/python3.3/site-packages/nikola/post.py", line 315, in compile
    self.is_two_file),
  File "/usr/lib/python3.3/site-packages/nikola/plugins/compile/rest/__init__.py", line 91, in compile_html
    }, logger=self.logger, l_source=source, l_add_ln=add_ln)
  File "/usr/lib/python3.3/site-packages/nikola/plugins/compile/rest/__init__.py", line 253, in rst2html
    pub.publish(enable_exit_status=enable_exit_status)
  File "/usr/lib/python3.3/site-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "/usr/lib/python3.3/site-packages/docutils/readers/__init__.py", line 72, in read
    self.parse()
  File "/usr/lib/python3.3/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/__init__.py", line 172, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/states.py", line 174, in run
    input_source=document['source'])
  File "/usr/lib/python3.3/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python3.3/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/states.py", line 2706, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/states.py", line 331, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/states.py", line 399, in new_subsection
    node=section_node, match_titles=True)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/states.py", line 286, in nested_parse
    node=node, match_titles=match_titles)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/states.py", line 199, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python3.3/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python3.3/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/states.py", line 2279, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/states.py", line 2291, in explicit_construct
    return method(self, expmatch)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/states.py", line 2034, in directive
    directive_class, match, type_name, option_presets)
  File "/usr/lib/python3.3/site-packages/docutils/parsers/rst/states.py", line 2083, in run_directive
    result = directive_instance.run()
  File "MY_BLOG/plugins/gallery_directive/gallery_directive.py", line 72, in run
    with open(gallery_index, 'r') as inf:
FileNotFoundError: [Errno 2] Datei oder Verzeichnis nicht gefunden: 'output/galleries/MY_GALLERY/index.html'
```

Rendering the galleries manually before the build fixes this:
```
# nikola render_galleries
# nikola build
```

**Comment by devurandom** [comment]
Persists with doit-0.27.0 and nikola-7.3.0:
```text
TaskError - taskid:render_posts:cache/posts/${MY_POST}.html
PythonAction Error
Traceback (most recent call last):
  File "${HOME}/.local/lib/python3.4/site-packages/doit/action.py", line 372, in execute
    returned_value = self.py_callable(*self.args, **kwargs)
  File "${HOME}/.local/lib/python3.4/site-packages/nikola/post.py", line 431, in compile
    self.is_two_file),
  File "${HOME}/.local/lib/python3.4/site-packages/nikola/plugins/compile/rest/__init__.py", line 99, in compile_html
    }, logger=self.logger, source_path=source, l_add_ln=add_ln)
  File "${HOME}/.local/lib/python3.4/site-packages/nikola/plugins/compile/rest/__init__.py", line 266, in rst2html
    pub.publish(enable_exit_status=enable_exit_status)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/readers/__init__.py", line 72, in read
    self.parse()
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/parsers/rst/__init__.py", line 172, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/parsers/rst/states.py", line 170, in run
    input_source=document['source'])
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/parsers/rst/states.py", line 2301, in explicit_markup
    self.explicit_list(blank_finish)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/parsers/rst/states.py", line 2331, in explicit_list
    match_titles=self.state_machine.match_titles)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/parsers/rst/states.py", line 319, in nested_list_parse
    node=node, match_titles=match_titles)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/parsers/rst/states.py", line 2604, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/parsers/rst/states.py", line 2311, in explicit_construct
    return method(self, expmatch)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/parsers/rst/states.py", line 2054, in directive
    directive_class, match, type_name, option_presets)
  File "${HOME}/.local/lib/python3.4/site-packages/docutils/parsers/rst/states.py", line 2103, in run_directive
    result = directive_instance.run()
  File "${MY_BLOG}/plugins/gallery_directive/gallery_directive.py", line 72, in run
    with open(gallery_index, 'r') as inf:
FileNotFoundError: [Errno 2] No such file or directory: 'output/galleries/${MY_GALLERY}/index.html'
```

**Comment by ralsina** [comment]
I finally have a way to fix this.

**Comment by ralsina** [comment]
Fixed in version 0.3 of the gallery_directive plugin. That version requires Nikola v7.4.1
